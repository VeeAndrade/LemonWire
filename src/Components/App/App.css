

.clapper {
  height: 80%;
  position: absolute;
}

.login-msg {
  float: right;
  width: 
}
