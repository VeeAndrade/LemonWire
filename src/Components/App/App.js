import React, { Component } from 'react';
import './App.css';
import MoviesContainer from '../MoviesContainer/MoviesContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      movies: []
    };
  }
  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(response => response.json())
      .then(movies => this.setState({movies: movies.movies}))
      .catch(error => console.log(error))
  }

  render() {
    return(
      <main className='main-app-display'>
        {this.state.movies.length ?
        <MoviesContainer movies={this.state.movies}/>
        : <p>Error</p>
      }
      </main>
    )
  }
}
export default App;
