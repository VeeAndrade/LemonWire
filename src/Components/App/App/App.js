import React, { Component } from 'react';
import './App.css';
import MoviesContainer from '../../MoviesContainer/MoviesContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      movies: []
    };
  }
  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(response => response.json())
      .then(movies => this.setState({movies: movies.movies}))
      .catch(error => console.log(error))
  }

  render() {
    return(
      <main>
        {this.state.movies.length ? 
        <MoviesContainer movies={this.state.movies}/>
        : <h3>Loading...</h3>
      }
      </main>
    )
  }
}
export default App;
