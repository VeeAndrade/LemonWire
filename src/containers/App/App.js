import React, { Component } from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import MoviesContainer from '../../containers/MoviesContainer/MoviesContainer';
import LoginForm from '../../containers/LoginForm/LoginForm'
import Nav from '../Nav/Nav';
import MovieDetails from '../MovieDetails/MovieDetails';
import Loading from '../../Components/Loading/Loading';
import './App.css';
import { connect } from 'react-redux'


export class App extends Component {
  render() {
    return(
       <main>
       <Switch>
         <Route exact path='/'>
           <Nav />
           <MoviesContainer />
         </Route>
         <Route path='/movies/:id'>
            {!this.props.user.id ? <Redirect to="/" /> : (
            <>
             <Nav />
             <MovieDetails />
            </>
           )}
         </Route>
         <Route exact path='/login'>
           {this.props.user.id ? <Redirect to='/' /> :
             <section className='login-page'>
               <h2 className='login-msg'>Get your ratings on.</h2>
               <img src={process.env.PUBLIC_URL + '/clapper.png'} alt='Clapperboard icon' className='clapper' />
               <LoginForm />
             </section>
           }
         </Route>
         <Redirect to='/' />
       </Switch>
      </main>
    )
  }
}

export const mapStateToProps = state => ({
  user: state.user,
  movies: state.movies
})

export default connect(mapStateToProps)(App);
