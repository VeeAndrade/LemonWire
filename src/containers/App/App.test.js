import React from 'react';
import ReactDOM from 'react-dom';
import  { App, mapStateToProps } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  describe('App container/component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<App movies={[{title: 'Frozen II'}]} user={{id:5}}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with a user and an array of movies', () => {
      const mockState = {
        movies: [ { name: 'Jumanji' } ],
        user: { name: 'Robbie' },
        ratings: [{rating: 5}, {rating: 4}],
      };
      const expected = {
        movies: [ { name: 'Jumanji' } ],
        user: { name: 'Robbie' },
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
    });
  });
});
