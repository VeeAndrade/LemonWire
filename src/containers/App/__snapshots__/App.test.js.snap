// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App container/component should match the snapshot 1`] = `
<main>
  <Route
    exact={true}
    path="/"
  >
    <Connect(Nav) />
    <Connect(MoviesContainer) />
  </Route>
  <Route
    exact={true}
    path="/movie_details"
  >
    <Connect(Nav) />
    <Connect(MovieDetails) />
  </Route>
  <Route
    exact={true}
    path="/login"
  >
    <Redirect
      to="/"
    />
  </Route>
</main>
`;
